android/src/main/java/com/imagepicker/ImagePickerModule.java

```diff
@@ -277,6 +275,7 @@ public void onActivityResult(int requestCode, int resultCode, Intent data) {
     }
 
     response.putString("uri", uri.toString());
+    response.putString("urlPath", realPath);
```

Comment:
> What kind of issues ?

App was crashing with large photos, if I remember correctly there's some sort of limit on how many bytes you can allocate on a RN string..?

> In your module you don't convert you filePath into data ?

I don't – that's why I ended up writing the upload in Java, eek :} My `@ReactMethod` in Java just receives the S3 url and the file path, and uploads the corresponding file using `HttpURLConnection` and a buffer.